src/Tronkell/Server/Server.hs

```diff
@@ -49,9 +50,9 @@ mainLoop server@Server{..} = do
 runClient :: Handle -> Server -> IO ()
 runClient clientHdl server@Server{..} = do
   hPutStr clientHdl "Take a nick name : "
-  nick <- cleanString <$> hGetLine clientHdl
-  let uId  = nickToUserId nick
-      user = User uId (Just . T.pack $ nick) Waiting
+  uId  <- modifyMVar serverLastUserId getNextUserID
+  nick <- Just . T.pack . cleanString <$> hGetLine clientHdl
```

Comment:
We'll need this for the next task, which is to make setting of the nick a client message (and thus a separate step from user creation).